fix: returning the correct email account not relying on 'default'

tukutela opened the pull request:
We hit an issue with sending emails using the Email Queue and Office365, what we noticed is that the email was received correctly from the right sender, however the email was never in the "Sent" folder of the person who sent it. A little further digging we found that ALL emails sent were in a default email "Sent" folder. 

The last update to ```frappe.email.smtp.py``` was to make a logic change to the condition that determines how to find the right ```Email Account```. See [PR 8368](https://github.com/frappe/frappe/pull/8368) - [Commit](https://github.com/frappe/frappe/pull/8368/commits/cfd7f8bf373d13ec02ea26ebb2db8b55c8b306b6). Its actually a good alteration however it has the effect of this method ALWAYS returning the ```frappe.local.outgoing_email_account["default"]``` as the ```frappe.local.outgoing_email_account[sender_email_id]``` never gets set. 

So I hereby submit this 2-line PR, which replaces the if statement that solves this problem.

saurabh6790 commented on frappe/email/smtp.py:
```diff
@@ -47,9 +47,8 @@ def get_outgoing_email_account(raise_exception_not_set=True, append_to=None, sen
-		or frappe.local.outgoing_email_account.get(sender_email_id)
-		or frappe.local.outgoing_email_account.get("default")):
+	if ((append_to and not frappe.local.outgoing_email_account.get(append_to)) or
+			(sender_email_id and not frappe.local.outgoing_email_account.get(sender_email_id))):
```
What if `append_to` and `sender email id` does not exist? Will, it returns default outgoing email account?

tukutela replied:
Yep, that’s the idea. Thanks for checking... Sorry, i didn't add this here. still learning github.

tukutela commented:
Yep, that’s the idea. Thanks for checking...

On Mon, 4 Nov 2019 at 9:11 pm, Saurabh <notifications@github.com> wrote:

> *@saurabh6790* requested changes on this pull request.
> ------------------------------
>
> In frappe/email/smtp.py
> <https://github.com/frappe/frappe/pull/8746#discussion_r341974536>:
>
> > @@ -47,9 +47,8 @@ def get_outgoing_email_account(raise_exception_not_set=True, append_to=None, sen
>  	if not getattr(frappe.local, "outgoing_email_account", None):
>  		frappe.local.outgoing_email_account = {}
>
> -	if not (frappe.local.outgoing_email_account.get(append_to)
> -		or frappe.local.outgoing_email_account.get(sender_email_id)
> -		or frappe.local.outgoing_email_account.get("default")):
> +	if ((append_to and not frappe.local.outgoing_email_account.get(append_to)) or
> +			(sender_email_id and not frappe.local.outgoing_email_account.get(sender_email_id))):
>
> What if append_to and sender email id does not exist? Will, it returns
> default outgoing email account?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/frappe/frappe/pull/8746?email_source=notifications&email_token=AACTGIB2WCC4NKI6NYVIKE3QR7YL7A5CNFSM4JIJXMZKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCKEQQXI#pullrequestreview-310970461>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AACTGIBLR6VLW43NOUEKWCLQR7YL7ANCNFSM4JIJXMZA>
> .
>
-- 
------------
Mark Walker
Maestro Results

tukutela commented:
hey @saurabh6790 

do you need anything from me? happy to help....

surajshetty3416 reviewed (changes requested):
@tukutela Please fix the failing tests in this PR.

tukutela commented:
OK, Updated as requested. A good learning experience.

Commit:
fix: serialize child table document objects

Commit:
fix: serialize datetime objects in child table documents

tukutela commented:
+1 adding activity to keep it current.

rohitwaghchaure committed:
fix: padding for progress bar in dashboard

rohitwaghchaure committed:
fix: not able to download XML file

Thunderbottom committed:
fix(security): invalidate reset_password_key on password reset

currently there is no way to invalidate reset_password_key on updating
password through the user settings. so whenever the user sets a new
password we'll invalidate the reset_password_key, so that existing links
to reset user passwords cannot be used.

Signed-off-by: Chinmay D. Pai <chinmaydpai@gmail.com>

mergify[bot] committed:
Merge pull request #8706 from hrwX/pref_global_search

fix: Missing results in global search

mergify[bot] committed:
Merge pull request #8982 from Thunderbottom/password-reset-fix

fix(security): invalidate reset_password_key on password reset

mergify[bot] committed:
Merge pull request #8906 from ruchamahabal/list_group_by_filter_fix

fix: list view group by filter ambiguous column name

mergify[bot] committed:
Merge pull request #8980 from rohitwaghchaure/fixed_xml_file_not_able_to_download_develop

fix: not able to download XML file

scmmishra committed:
style: added semi-colon

saurabh6790 committed:
fix: add hourly long frequency to scheduled job type

scmmishra committed:
feat: summary tag should have cursor pointer

mergify[bot] committed:
Merge pull request #8993 from saurabh6790/scheduled_job_type_frequency

fix: add hourly long frequency to scheduled job type

ruchamahabal committed:
fix: remove illustrations from onboarding slides

ruchamahabal committed:
fix: set the modal backdrop opacity to 1

mergify[bot] committed:
Merge pull request #8969 from Alchez/dev-sign-up-role

fix: add default role on sign up via SSO

mergify[bot] committed:
Merge pull request #8994 from scmmishra/summary-details

fix: summary tag should have cursor pointer

mergify[bot] committed:
Merge pull request #8949 from scmmishra/chart-opts-query-report

fix: query report chart options

rmehta committed:
fix: make padding a bit more consistent

ruchamahabal committed:
fix: reduced modal's box-shadow and added a border

prssanna committed:
fix(tests): add UI test for next previous document navigation

Thunderbottom committed:
fix(email): try to encode email part to utf-8 (#8965)

fixes issues with class objects other than str

Traceback (most recent call last):
  File "/home/frappe/frappe-bench/apps/frappe/frappe/email/doctype/email_account/email_account.py", line 281, in receive
    communication = self.insert_communication(msg, args=args)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/email/doctype/email_account/email_account.py", line 338, in insert_communication
    email = Email(raw)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/email/receive.py", line 372, in __init__
    self.parse()
  File "/home/frappe/frappe-bench/apps/frappe/frappe/email/receive.py", line 393, in parse
    self.process_part(part)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/email/receive.py", line 445, in process_part
    self.text_content += self.get_payload(part)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/email/receive.py", line 489, in get_payload
    charset = self.get_charset(part)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/email/receive.py", line 484, in get_charset
    charset = chardet.detect(frappe.safe_encode(part))['encoding']
  File "/home/frappe/frappe-bench/env/lib/python3.6/site-packages/chardet/__init__.py", line 34, in detect
    '{0}'.format(type(byte_str)))
TypeError: Expected object of type bytes or bytearray, got: <class 'email.message.Message'>

Signed-off-by: Chinmay D. Pai <chinmaydpai@gmail.com>

Mangesh-Khairnar committed:
fix(autocomplete): parse options (#8998)

* fix(autocomplete): parse options

* fix: add ignore validation flag

* fix: minor change

mergify[bot] committed:
Merge pull request #8963 from prssanna/next-doc-fix

fix: Ambiguous column error when going to next doc

Thunderbottom committed:
fix(patch): auto commit on more than 10000 writes

fixes issue where patch fails with:

frappe.exceptions.ValidationError: Too many writes in one request. Please send smaller requests

Signed-off-by: Chinmay D. Pai <chinmaydpai@gmail.com>

tukutela commented:
bump

surajshetty3416 committed:
fix: Allow field of submitted doc to edit if field has allow_on_submit enabled

scmmishra committed:
fix: permission error

prssanna committed:
fix: reopen todo only when assignment rule condition is satisfied

Mangesh-Khairnar committed:
fix(scheduler): import update site config

prssanna committed:
fix: reopen all todos if close condition is not satisfed

nextchamp-saqib committed:
revert: changes in currency formatting (#8996)

* fix: reverted currency formatting

* ux: added description for currency's number format
* It doesn't have any effect on how the currency is formatted.

* fix: revert json changes in currency master

shridarpatil committed:
Bug fixed ID value should not be set to 0 by default

mergify[bot] committed:
Merge pull request #9012 from Mangesh-Khairnar/fix-scheduler

fix(scheduler): import update site config

Thunderbottom commented:
@tukutela looks good, but we need some tests for this

mergify[bot] committed:
Merge pull request #9009 from frappe/perm-error

fix: permission error

surajshetty3416 committed:
fix: Check if the type of row data is tuple

surajshetty3416 committed:
test: Add test to check if field is editable

surajshetty3416 committed:
fix: Typo

mergify[bot] committed:
Merge pull request #9015 from zerodhatech/develop

fix(filters): set default id to empty string instead of 0

rohitwaghchaure committed:
Merge pull request #8974 from rohitwaghchaure/fixed_css_for_dashboard_progress_bar

fix: padding for progress bar in dashboard

Thunderbottom committed:
fix(chat): refactor code to make chat work

fixes issues with token being wrongly passed as the owner, instead of
current session user (ie. frappe.session.user), which in turn causes the
following issue:

Traceback (most recent call last):
  File "/home/frappe/frappe-bench/apps/frappe/frappe/app.py", line 57, in application
    response = frappe.handler.handle()
  File "/home/frappe/frappe-bench/apps/frappe/frappe/handler.py", line 22, in handle
    data = execute_cmd(cmd)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/handler.py", line 61, in execute_cmd
    return frappe.call(method, **frappe.form_dict)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/__init__.py", line 1042, in call
    return fn(*args, **newargs)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/chat/doctype/chat_room/chat_room.py", line 203, in create
    room.save(ignore_permissions = True)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/model/document.py", line 272, in save
    return self._save(*args, **kwargs)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/model/document.py", line 295, in _save
    self.insert()
  File "/home/frappe/frappe-bench/apps/frappe/frappe/model/document.py", line 223, in insert
    self._validate_links()
  File "/home/frappe/frappe-bench/apps/frappe/frappe/model/document.py", line 756, in _validate_links
    frappe.LinkValidationError)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/__init__.py", line 364, in throw
    msgprint(msg, raise_exception=exc, title=title, indicator='red')
  File "/home/frappe/frappe-bench/apps/frappe/frappe/__init__.py", line 350, in msgprint
    _raise_exception()
  File "/home/frappe/frappe-bench/apps/frappe/frappe/__init__.py", line 316, in _raise_exception
    raise raise_exception(msg)
frappe.exceptions.LinkValidationError: Could not find Owner: 0a4d6eb7763ef2065f44ece0653d829a1ba189f753562ab9ac261620

also fixes issue where the chat rooms do not load because of wrongly
passed parameters

Signed-off-by: Chinmay D. Pai <chinmaydpai@gmail.com>

prssanna committed:
fix(Leaderboard): Leaderboard timespan options (#8953)

* fix: change leaderboard timespan options and add select from date option

* fix: leaderboard timespan options grouping and add quarter_start and quarter_end to datetime.js

* fix: codacy

* fix: add last month

Thunderbottom committed:
fix(chat): make keyword changes to chat.js

Signed-off-by: Chinmay D. Pai <chinmaydpai@gmail.com>

Thunderbottom committed:
fix(chat): change sql formatting

Signed-off-by: Chinmay D. Pai <chinmaydpai@gmail.com>

mergify[bot] committed:
Merge pull request #9008 from Thunderbottom/auto-commit-prepared-rep

fix(patch): auto commit on more than 10000 writes

surajshetty3416 committed:
fix: Hidden columns should be hidden even after refresh

scmmishra committed:
fix: height for null state on list view (#8997)

* fix: height for null state on list view

* Update list.less

oshosanya committed:
feat: Allow full database config on creating site (#8874)

* Allow setting db port on install

* Allow setting db port on install

* Allow full db config on creating site

* bug fix

* bug fix

* Escape character

* Update frappe/commands/site.py

allow only integer argument

Co-Authored-By: Chinmay Pai <chinmaydpai@gmail.com>

* Update frappe/database/db_manager.py

No need to escape integer variable

Co-Authored-By: Chinmay Pai <chinmaydpai@gmail.com>

* cast port to integer to surpress pymysql error

* cast port to string to surpress pymysql error

* Removed type casting

khushalti committed:
fix: link field search taking trailing and leading spaces (#9031)

surajshetty3416 committed:
fix(file): Public-private issue (#9029)

* fix: Pass file's is_private for doc creation

* fix: File is_private and file_url mismatch

* fix: Check if file_url exists

* fix: Validation check

* fix(postgres): Convert is_private to int

* fix: File path for content_hash

deepeshgarg007 committed:
Merge barnch develop into mandatory-depends-on

hrwX committed:
feat: add default email account for email linking (#8833)

* feat: add default email account for email linking

* fix: remove unused import

* fix: remove try-catch

deepeshgarg007 committed:
fix: Add mandatory depends on and read only depends on field in custom field doctype

surajshetty3416 committed:
fix: Add a patch to fix all user's home settings

Thunderbottom committed:
fix: use filter for removing administrator from email list

Signed-off-by: Chinmay D. Pai <chinmaydpai@gmail.com>
Co-authored-by: Saurabh Palande <saurabh6790@gmail.com>

prssanna committed:
fix: grid file upload fix

mergify[bot] committed:
Merge pull request #9038 from prssanna/upload-fix

fix: grid file upload fix

prssanna committed:
fix(Grid): Child table pagination fixes (#9034)

* fix: don't render result if grid_rows is undefined and fix truncate_rows

* fix: change page length to 50

* fix: grid pagination ui test

* fix: confirm before deleting all child table rows

surajshetty3416 committed:
fix: Show default currency as a fallback for currency fieldtype (#8944)

* fix: Show default currency as a fallback for currency fieldtype

* test: Add a test for currency formatting

surajshetty3416 committed:
fix: Remove limit

mergify[bot] committed:
Merge pull request #9026 from surajshetty3416/fix-report-column-hide

fix(report): Hidden columns should be hidden even after refresh

mergify[bot] committed:
Merge pull request #9037 from surajshetty3416/fix-desk-modules

fix: Add a patch to fix all user's home settings

surajshetty3416 committed:
test: Add assert statement for value

surajshetty3416 committed:
Merge branch 'develop' into fix-report-view-edit-option

mergify[bot] committed:
Merge pull request #9016 from surajshetty3416/fix-report-user-permission

fix: User permission issue in query report

surajshetty3416 committed:
test: Clear cache after creating doctype

surajshetty3416 committed:
Merge pull request #9011 from surajshetty3416/fix-report-view-edit-option

fix: Allow field of submitted doc to edit if field has allow_on_submit enabled

vishdha committed:
clear attachment on image remove attach file

surajshetty3416 committed:
fix: Empty row validation

surajshetty3416 committed:
style: Fix code formatting

mergify[bot] committed:
Merge pull request #9044 from surajshetty3416/fix-grid

fix(grid): Empty row validation

prssanna committed:
fix: before setting user default value, if fieldtype is Link check if doc exists

prssanna committed:
fix: don't validate email for standard filters

mergify[bot] committed:
Merge pull request #9047 from prssanna/filter-email-fix

fix: don't validate email for standard filters

mergify[bot] committed:
Merge pull request #9046 from prssanna/new-doc-fix

fix: user default value for Link fields

cjpit committed:
fix: rfc5322 compliance for python3 (#8912)

* start of tests

* rfc compliant emails for frappe.

* fix for deepsource

* fix length of strings.

* fix bug that I'm not sure where it came from?

* codacy and deepsource issues trying to keep them happy

* take a punt in the dark

* fix: use SMTPUTF8 instead of SMTP and other minor fixes

Signed-off-by: Chinmay D. Pai <chinmaydpai@gmail.com>

prssanna committed:
fix: no records tagged getting rendered multiple times

prssanna committed:
fix: add padding to 'No Records Tagged'

prssanna committed:
fix: use reload_stats function

surajshetty3416 committed:
feat (#9052)

(server-scripts): Support Before/After Save for submitted document

ci2014 committed:
Treat html editor as data field in standard filter

HTML Editor field can be treated as normal data field and be filtered with Data fieldtype and condition 'like'

surajshetty3416 committed:
Revert "fix: user default value for Link fields"

surajshetty3416 committed:
Revert "fix: user default value for Link fields" (#9061)

Revert "fix: user default value for Link fields"

prssanna committed:
before setting user default value, if fieldtype is Link check if doc exists

surajshetty3416 committed:
fix: Return empty dict if session is not created

surajshetty3416 committed:
fix: Remove unwanted print statement

surajshetty3416 committed:
fix: user default value for Link fields (#9062)

fix: user default value for Link fields

surajshetty3416 committed:
fix: Get guest's config if session.user is None

vishdha committed:
fix: Remove Attachment on remove

vishdha committed:
fix: minor changes

surajshetty3416 committed:
style: fix Codacy issue

There should be no space before ','. (comma-spacing)

mergify[bot] committed:
Merge pull request #9066 from vishdha/remove_attachment

fix: on image remove event remove attach file.

prssanna committed:
fix: list sidebar tags dropdown doesn't get updated

prssanna committed:
fix: fix tag dropdown search

surajshetty3416 committed:
Merge branch 'develop' into remove_onboarding_images

Thunderbottom committed:
feat(email): allow ssl for outgoing emails, append to sent folder

* dont always use default outgoing account for sending
* allow ssl for outgoing emails

Signed-off-by: Chinmay D. Pai <chinmaydpai@gmail.com>

surajshetty3416 committed:
chore: Add comment to explain the reason for ignoring validation

mergify[bot] committed:
Merge pull request #8927 from hrwX/report_tag_fix

fix(Report): ignore args if Tag Link

prssanna committed:
fix: fix grid ui for child tables in column break (#9050)

hrwX committed:
fix: miscellaneous fixes

surajshetty3416 committed:
fix: Make "Restriction" text translatable

surajshetty3416 committed:
fix: Modal indicator CSS

surajshetty3416 committed:
fix: Restricted button CSS

mergify[bot] committed:
Merge pull request #9073 from surajshetty3416/fix-rtl-css

fix: Make "Restriction" text translatable and fix style

rohitwaghchaure committed:
fix: patch home_settings_for_all_users

mergify[bot] committed:
Merge pull request #9077 from rohitwaghchaure/fixed_patch_home_settings_for_users_develop

fix: patch home_settings_for_all_users

surajshetty3416 committed:
fix: List filter area style

surajshetty3416 committed:
Merge pull request #9081 from surajshetty3416/fix-list-filter-area-style

fix: List filter area style

surajshetty3416 committed:
fix: KeyError: 'hidden_modules'

surajshetty3416 committed:
fix: KeyError: 'hidden_modules' while migrating (#9084)

fix: KeyError: 'hidden_modules' while migrating

surajshetty3416 committed:
fix: TypeError in home_settings patch

Convert set to list

hrwX committed:
fix: rename first_name

mergify[bot] committed:
Merge pull request #9087 from surajshetty3416/fix-patch-error

fix: TypeError in home_settings patch

mergify[bot] committed:
Merge pull request #9013 from prssanna/todo-fix

fix: ToDo reopens even when status is Closed

Mangesh-Khairnar committed:
fix: allow to format currency for guest users

scmmishra committed:
refactor: override style for changelog modal

mergify[bot] committed:
Merge pull request #8970 from scmmishra/changelog-ux

fix (ui/ux): change log modal on boot

mergify[bot] committed:
Merge pull request #9002 from ruchamahabal/remove_onboarding_images

fix: remove illustrations from onboarding slides

surajshetty3416 committed:
test: Fix form's randomly failing test

surajshetty3416 committed:
fix(login): Get guest's config if session.user is None (#9101)

fix(login): Get guest's config if session.user is None

surajshetty3416 committed:
fix: Fix tags list sidebar dropdown (#9099)

fix: Fix tags list sidebar dropdown

Co-authored-by: Prssanna Desai <prssud@gmail.com>

surajshetty3416 committed:
Merge pull request #9102 from surajshetty3416/fix-form-test

test: Fix randomly failing test in form.js

surajshetty3416 committed:
Merge branch 'develop' into v12-pre-release-fixes-develop

surajshetty3416 committed:
Merge branch 'develop' into tag-fix-develop

surajshetty3416 committed:
fix: Ignore permissions for tags (#9100)

fix: Ignore permissions for tags

Co-authored-by: Himanshu <himanshuwarekar@yahoo.com>

surajshetty3416 committed:
Merge branch 'develop' into tag-fix-develop

surajshetty3416 committed:
test: Fix failing test in form.js (#9103)

surajshetty3416 committed:
Merge branch 'develop' into tag-fix-develop

gavindsouza committed:
fix: bench build "Cannot link {assets} to {site assets}"

surajshetty3416 committed:
fix: list sidebar tags dropdown doesn't get updated (#9098)

fix: list sidebar tags dropdown doesn't get updated

Co-authored-by: Prssanna Desai <prssud@gmail.com>

rmehta committed:
feat(safe_exec): allow iterators and comprehension

surajshetty3416 committed:
Merge version-12-hotfix to develop (#9095)

* fix: REST API utf-8 decoding on creates/updates

Creating or updating a document via the REST API would generate an error of:

`TypeError: the JSON object must be str, not 'bytes'`

Because get_data() returns bytes which must be explicitly converted to a string before parsing as JSON.

Defect introduced by efe94886a and a71a92341e

* fix(patch): Handle failures in while making Prepared Report attachments private

* case insensitive search for postgres

* convert operator to lowercase while checking

* fix: Pass prepared_report_name as filter if exists

* fix: Add input to multiselect_list control

* fix(reportview): convert to unicode conditionally

fixes TypeError: decoding str is not supported

Traceback (most recent call last):
  File "/home/frappe/frappe-bench/apps/frappe/frappe/app.py", line 57, in application
    response = frappe.handler.handle()
  File "/home/frappe/frappe-bench/apps/frappe/frappe/handler.py", line 22, in handle
    data = execute_cmd(cmd)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/handler.py", line 61, in execute_cmd
    return frappe.call(method, **frappe.form_dict)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/__init__.py", line 1038, in call
    return fn(*args, **newargs)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/__init__.py", line 511, in wrapper_fn
    retval = fn(*args, **get_newargs(fn, kwargs))
  File "/home/frappe/frappe-bench/apps/frappe/frappe/desk/reportview.py", line 177, in export_query
    frappe.response['result'] = text_type(f.read(), 'utf-8')
TypeError: decoding str is not supported

Signed-off-by: Chinmay D. Pai <chinmaydpai@gmail.com>

* chore: remove useless encode

Signed-off-by: Chinmay D. Pai <chinmaydpai@gmail.com>

* fix: Allow Rename in Website Route Meta

* fix: Add Visit Web Page button in form

* fix: next schedule date should be on or after current date

* test: next schedule date

* fix: Change modified by

* fix: do not allow bulk update for core, single doctypes and doctypes from inactive domains

* fix: switch to safe_decode

* test: Fix list_view test by removing clear-cache code (#8941)

* test: Fix listview test by waiting for clear-cache

* test: Move wait to list_view.js

* test: Try removing clear-cache

* fix(integration): Social Login Key (#8940)

add missing field

* fix: don't set filter for restrict_to_domain in Bulk Update Tool

* fix: query report chart options

* fix: Default and company address fixes

* fix: revert changes made to get_default_address function

* fix: Remove changes in notifications

* fix: list view group by filter ambiguous column name

* fix: changelog modal ui

* test: list view child table filter with created by filter

* fix: not able to download XML file

* fix(security): invalidate reset_password_key on password reset

currently there is no way to invalidate reset_password_key on updating
password through the user settings. so whenever the user sets a new
password we'll invalidate the reset_password_key, so that existing links
to reset user passwords cannot be used.

Signed-off-by: Chinmay D. Pai <chinmaydpai@gmail.com>

* fix: Pin faker version to avoid test failures

* perf: optimise globals search

* fix: add index on child table

* fix: do not pop item from list

* fix: dont add index for global search doctype

* fix: rename function to set

* tests: fix results being return

* fix: codacy fixes

* fix: Code cleanup

- Pass values so that db cursor can handle escaping

* fix: Convert list to tuple

* style: added semi-colon

* fix: add default role on sign up via SSO

* fix(autocomplete): parse options (#8999)

* fix(autocomplete): parse options

* fix: add ignore validation flag

* fix: minor change

* fix(email): try to encode email part to utf-8 (#8964)

fixes issues with class objects other than str

Traceback (most recent call last):
  File "/home/frappe/frappe-bench/apps/frappe/frappe/email/doctype/email_account/email_account.py", line 281, in receive
    communication = self.insert_communication(msg, args=args)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/email/doctype/email_account/email_account.py", line 338, in insert_communication
    email = Email(raw)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/email/receive.py", line 372, in __init__
    self.parse()
  File "/home/frappe/frappe-bench/apps/frappe/frappe/email/receive.py", line 393, in parse
    self.process_part(part)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/email/receive.py", line 445, in process_part
    self.text_content += self.get_payload(part)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/email/receive.py", line 489, in get_payload
    charset = self.get_charset(part)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/email/receive.py", line 484, in get_charset
    charset = chardet.detect(frappe.safe_encode(part))['encoding']
  File "/home/frappe/frappe-bench/env/lib/python3.6/site-packages/chardet/__init__.py", line 34, in detect
    '{0}'.format(type(byte_str)))
TypeError: Expected object of type bytes or bytearray, got: <class 'email.message.Message'>

Signed-off-by: Chinmay D. Pai <chinmaydpai@gmail.com>

* fix: Ambiguous column error when going to next doc (#9005)

fix: Ambiguous column error when going to next doc

Co-authored-by: Prssanna Desai <prssud@gmail.com>

* fix(patch): auto commit on more than 10000 writes

fixes issue where patch fails with:

frappe.exceptions.ValidationError: Too many writes in one request. Please send smaller requests

Signed-off-by: Chinmay D. Pai <chinmaydpai@gmail.com>

* revert: changes in currency formatting  (#9003)

* fix: reverted currency formatting

* ux: added description for currency's number format
* It doesn't have any effect on how the currency is formatted.

* fix: revert json changes in currency master

* fix(filters): set default id to empty string instead of 0 (#9014)

fixes issue where "undefined" value gets set to 0 in postgres, which causes ProgrammingError since the expected value is of type str and not int

* revert: changes in currency formatting  (#9003)

* fix: reverted currency formatting

* ux: added description for currency's number format
* It doesn't have any effect on how the currency is formatted.

* fix: revert json changes in currency master

* feat: add doc for email linking

* fix: missing commits while reverting currency changes

* fix(file): Public-private issue (#9032)

* fix: Pass file's is_private for doc creation

* fix: File is_private and file_url mismatch

* fix: Check if file_url exists

* fix: Validation check

* fix(postgres): Convert is_private to int

* fix: File path for content_hash

* fix: Show default currency as a fallback for currency fieldtype (#8992)

* fix: Show default currency as a fallback for currency fieldtype

* test: Add a test for currency formatting

* fix: Add a patch to fix all user's home settings (#9040)

* fix: Add a patch to fix all user's home settings

* fix: Remove limit

* fix: Allow field of submitted doc to edit if field has allow_on_submit enabled

* test: Add test to check if field is editable

* fix: Typo

* test: Add assert statement for value

* test: Clear cache after creating doctype

* fix: Codacy

* fix checkbox behavior in rtl view when clicking on label to check (#8898)

* fix checkbox behavior in rtl view when clicking on label to check

* fix: Undo changes in desk.css

* fix: Empty row validation

* style: Fix code formatting

* fix: Terminate regex

* fix: don't validate email for standard filters

* fix: before setting user default value, if fieldtype is Link check if doc exists

* Revert "fix: user default value for Link fields"

* before setting user default value, if fieldtype is Link check if doc exists

* fix: Remove Attachment on remove

* fix: minor changes

* style: fix Codacy issue

There should be no space before ','. (comma-spacing)

* fix: rfc5322 compliance for python3 (#9056)

* start of tests

* rfc compliant emails for frappe.

* fix for deepsource

* fix length of strings.

* fix bug that I'm not sure where it came from?

* codacy and deepsource issues trying to keep them happy

* take a punt in the dark

* fix: use SMTPUTF8 instead of SMTP and other minor fixes

Signed-off-by: Chinmay D. Pai <chinmaydpai@gmail.com>

* fix: patch home_settings_for_all_users

* style: Fix Codacy

* fix: KeyError: 'hidden_modules'

* fix: TypeError in home_settings patch

Convert set to list

* refactor: override style for changelog modal

* Merge V12-pre-release to version-12-hotfix (#9094)

Merge V12-pre-release to version-12-hotfix

Co-authored-by: null <rohitw1991@gmail.com>
Co-authored-by: Prssanna Desai <prssud@gmail.com>
Co-authored-by: Chinmay Pai <chinmaydpai@gmail.com>
Co-authored-by: Vishal Dhayagude <vishaldhayagude09@gmail.com>
Co-authored-by: Himanshu <himanshuwarekar@yahoo.com>
Co-authored-by: null <37929162+mergify[bot]@users.noreply.github.com>
Co-authored-by: null <sahilkhan28297@gmail.com>

* Revert queue.py

Co-authored-by: Ben Knowles <benknowles@users.noreply.github.com>
Co-authored-by: Aditya Hase <aditya@adityahase.com>
Co-authored-by: Shridhar Patil <shridharpatil2792@gmail.com>
Co-authored-by: Faris Ansari <netchampfaris@users.noreply.github.com>
Co-authored-by: Chinmay Pai <chinmaydpai@gmail.com>
Co-authored-by: Rucha Mahabal <ruchamahabal2@gmail.com>
Co-authored-by: Revant Nandgaonkar <revant.one@gmail.com>
Co-authored-by: mergify[bot] <37929162+mergify[bot]@users.noreply.github.com>
Co-authored-by: Shivam Mishra <scmmishra@users.noreply.github.com>
Co-authored-by: Deepesh Garg <42651287+deepeshgarg007@users.noreply.github.com>
Co-authored-by: rohitwaghchaure <rohitw1991@gmail.com>
Co-authored-by: Himanshu <himanshuwarekar@yahoo.com>
Co-authored-by: Rohan <Alchez@users.noreply.github.com>
Co-authored-by: Mangesh-Khairnar <mkhairnar10@gmail.com>
Co-authored-by: Prssanna Desai <prssud@gmail.com>
Co-authored-by: Saqib <thefalconx33@gmail.com>
Co-authored-by: Nabin Hait <nabinhait@gmail.com>
Co-authored-by: Ammar Hararah <ammarkharara@gmail.com>
Co-authored-by: Vishal Dhayagude <vishdha@users.noreply.github.com>
Co-authored-by: sahil28297 <37302950+sahil28297@users.noreply.github.com>

mergify[bot] committed:
Merge pull request #9096 from Mangesh-Khairnar/fix-format-currency

fix: allow to format currency for guest users

rmehta committed:
fix(safe_exec): allow frappe.utils.[function]

surajshetty3416 committed:
Merge branch 'develop' into mandatory-depends-on

rmehta committed:
fix(safe_exec): allow sorted

gavindsouza committed:
fix: avoid race condition to create symlinks

surajshetty3416 committed:
fix: Treat html editor as data field in standard filter (#9108)

fix: Treat html editor as data field in standard filter

Co-authored-by: null <christoph.isik@wpci.de>

rohitwaghchaure committed:
fix: incorrect parameter passing

nextchamp-saqib committed:
fix: display correct currency in child tables

netchampfaris committed:
fix: Whitelist schema.org attributes

so that they can be used in HTML Editor in web pages

prssanna committed:
fix: only scroll to top when going to a new page to add a row

prssanna committed:
fix: don't create filter tag if no filter is applied

rohitwaghchaure committed:
Merge pull request #9110 from rohitwaghchaure/incorrect_parameter_passing_develop

fix: incorrect parameter passing

surajshetty3416 committed:
test: Increase defaultCommandTimeout to 10sec (#9115)

* test: Fix failing test in form.js

* test: Increase defaultCommandTimeout to 10sec

hrwX committed:
fix: distinct tag total

surajshetty3416 committed:
Merge branch 'develop' into filter-fix

mergify[bot] committed:
Merge pull request #9120 from prssanna/filter-fix

fix: don't create filter tag if no filter is applied

surajshetty3416 committed:
Merge branch 'develop' into tag_total_develop

scmmishra committed:
perf: Speed improvements for moduleview get (#9079)

* feat: api to setup boot cache on login

* feat (perf): use cache data instead of query

* feat: added doc hooks for rebuilding cache

* feat: improved can import query

* fix: ignore cache rebuild for import or install

* fix: fix postgres syntax error

Co-authored-by: Suraj Shetty <13928957+surajshetty3416@users.noreply.github.com>

rmehta committed:
feat(safe_exec): allow frappe.db.set_value

surajshetty3416 committed:
Merge branch 'develop' into tag_total_develop

rmehta committed:
fix(minor): remove print from safe_exec

netchampfaris committed:
feat: Custom CSS support in Web Form

mergify[bot] committed:
Merge pull request #9114 from prssanna/grid-fix

fix: only scroll to top when going to a new page to add a row

surajshetty3416 committed:
Merge branch 'develop' into tag_total_develop

netchampfaris committed:
fix(form-nav): Get filters from list_view (#9124)

* fix(form-nav): Get filters from list_view

- Fallback to user_settings
- Fix UI Test

* style: Missing semicolon

mergify[bot] committed:
Merge pull request #9121 from netchampfaris/web-form-custom-css

feat: Custom CSS support in Web Form

hrwX committed:
fix: ignore distinct for tag link

hrwX committed:
Merge branch 'tag_total_develop' of https://github.com/hrwx/frappe into tag_total_develop

rmehta committed:
Merge pull request #9105 from rmehta/safe-exec-allow-iter

feat(safe_exec): allow iterators and comprehension

hrwX committed:
Update frappe/desk/reportview.py

Co-Authored-By: Suraj Shetty <13928957+surajshetty3416@users.noreply.github.com>

hrwX committed:
Update frappe/desk/reportview.py

Co-Authored-By: Suraj Shetty <13928957+surajshetty3416@users.noreply.github.com>

hrwX committed:
Update frappe/desk/reportview.py

Co-Authored-By: Suraj Shetty <13928957+surajshetty3416@users.noreply.github.com>

hrwX committed:
Update frappe/desk/reportview.py

Co-Authored-By: Suraj Shetty <13928957+surajshetty3416@users.noreply.github.com>

hrwX committed:
Update frappe/desk/reportview.py

Co-Authored-By: Suraj Shetty <13928957+surajshetty3416@users.noreply.github.com>

hrwX committed:
Update frappe/desk/reportview.py

Co-Authored-By: Suraj Shetty <13928957+surajshetty3416@users.noreply.github.com>

hrwX committed:
review changes

mergify[bot] committed:
Merge pull request #9117 from hrwX/tag_total_develop

fix(Tag): distinct tag total

surajshetty3416 committed:
Merge branch 'develop' into whitelist-schemaorg-attributes

mergify[bot] committed:
Merge pull request #9112 from netchampfaris/whitelist-schemaorg-attributes

fix: Whitelist schema.org attributes

hrwX committed:
fix: use distinct for tag filters (#9126)

netchampfaris committed:
fix: Update in New Data Import

- Build a doc from the provided columns in template
- Check if ID column is provided

mergify[bot] committed:
Merge pull request #9127 from netchampfaris/fix-update-in-new-data-import

fix: Update in New Data Import

netchampfaris committed:
fix: Child Table Update in New Data Import

- Dont check for mandatory in child rows in update
- Create a new doc for child rows being inserted in update

surajshetty3416 committed:
Merge branch 'develop' into table-currency-pf

mergify[bot] committed:
Merge pull request #9129 from netchampfaris/fix-new-data-import-update-for-child-rows

fix: Child Table Update in New Data Import

gbm001 committed:
feat: Optionally remove seconds from datetime (#8531)

* fix: Add updated datepicker; fixed seconds formatting bug.
Seconds between 0 and 9 were not zero-padded.

* feat: Add framework for time format

* feat: datetime server-side formatters.

* tests: Added server-side datetime formatter tests

* feat: Update client-side datetime formatters

* tests: Add Cypress client-side formatting tests.

* fix: JSON errors

* fix: Update to not hard-code admin password

* fix: Change to using bulk_update rather than the REST API

* tests: Use Custom doctype for testing, not Standard

* fix: Codacy style fixes

* fix: Commonify update_datetime_picker in date.js, datetime.js, time.js
Fix order of time_format in System Settings
Restore get_user_fmt in utils/datetime.js

* feat: Drastically reduce scale of Cypress testing (to make tests faster)
Full testing is possible by setting 'fast_mode' to false in the spec file.

* fix: Fix issues with datepicker/timepicker expansion

* fix: typo

* style: Various style fixes as requested by DeppSource: Python

* fix: Timepicker not hiding on 'now' button. Force hiding on click.

* style: Codacy style fixes.

* fix: Use datepicker from node_modules

* test: Refactor Datetime UI tests

- cy.get_field
- cy.set_value
- cy.insert_doc with ignore_duplicate
- Nominal datetime tests to cover most formats
- Formatting with prettier

* test: Datetime UI tests; wait for cur_frm.doc.datetime to update

* tests: Add whitespace to typed input

- Clear input only for Time field

* test: Wait timeout 200

* test: Fix form test

Co-authored-by: Faris Ansari <netchampfaris@users.noreply.github.com>

mergify[bot] committed:
Merge pull request #9111 from nextchamp-saqib/table-currency-pf

fix: display correct currency in child tables

gavindsouza committed:
fix: python 2 compatibility for symlink creation

deepeshgarg007 committed:
Merge branch 'develop' of https://github.com/frappe/frappe into mandatory-depends-on

Commit:
fix(pdf): error when trying to print PDFs

developmentforpeople committed:
Fix non translatable text for edit/delete comments in doctypes timeline

prssanna committed:
fix: whitelist function

Co-Authored-By: Shivam Mishra <scmmishra@users.noreply.github.com>

mergify[bot] committed:
Merge pull request #9138 from developmentforpeople/fix-timeline-translatable-text

fix: Fix non translatable text for edit/delete comments in doctypes timeline

Commit:
fix: no data fallback on filters

scmmishra committed:
feat: wider awesomebar

hrwX committed:
fix: adding quick entry fields via customize form

surajshetty3416 committed:
fix: File duplicate check (#9136)

* fix: File duplicate check

By passing file name from file url instead of file name.
because file name can be different from actual file
if the same file is uploaded twice with different names.

* refactor: Remove unused code

mergify[bot] committed:
Merge pull request #9143 from Alchez/dev-filters-default-message

fix: no data fallback on filters

Thunderbottom committed:
fix(chat): separate keywords for user and token

Signed-off-by: Chinmay D. Pai <chinmaydpai@gmail.com>

Thunderbottom committed:
chore: fix indentation and cleanup annoying spaces

Signed-off-by: Chinmay D. Pai <chinmaydpai@gmail.com>

sagarvora committed:
fix: update version pinning for python 3.8 compatibility

scmmishra committed:
fix: formatting for check in prints (#9139)

* fix: formatting for check in prints

* refactor: remove important

Co-authored-by: Suraj Shetty <13928957+surajshetty3416@users.noreply.github.com>

mergify[bot] committed:
Merge pull request #8820 from prssanna/mandatory-depends-on

feat: Add Mandatory Depends On and Read Only Depends On to Docfield

Thunderbottom committed:
fix(email): handle case where cstr returns text_type of str

chardet requires input to be bytes or bytesarray, but sometimes
frappe.cstr() returns text_type of str without encoding it to utf-8

Signed-off-by: Chinmay D. Pai <chinmaydpai@gmail.com>

mergify[bot] committed:
Merge pull request #9149 from sagarvora/fix-req

fix: update version pinning for python 3.8 compatibility

surajshetty3416 committed:
fix(email): Do not encode smtp_server value (#9137)

* fix: Do not encode smtp server

Because it breaks in python 3

* fix: use cstr to change text_type to str

Signed-off-by: Chinmay D. Pai <chinmaydpai@gmail.com>

Co-authored-by: Chinmay Pai <chinmaydpai@gmail.com>

mergify[bot] committed:
Merge pull request #9135 from Alchez/dev-os-error-message

fix(pdf): error when trying to print PDFs

prssanna committed:
fix: hide Cancel button if workflow state and action for Cancel… (#9151)

sagarvora committed:
fix: (re)allow custom filters/methods to be used with jinja

gavindsouza committed:
feat: show_alert flag in rename_doc API

deepeshgarg007 committed:
fix: Fallback value for time format

mergify[bot] committed:
Merge pull request #9166 from deepeshgarg007/date_fix

fix: Fallback value for time format

rohitwaghchaure committed:
Merge pull request #8775 from rohitwaghchaure/email_not_pull_because_of_contact_error

fix: Name of Contact cannot be Contact error while creating communication for incoming emails

Commit:
fix: allow tables to be sent in webhook

surajshetty3416 committed:
Merge branch 'develop' into awesome-bar

surajshetty3416 committed:
Merge branch 'develop' into bytes-fix

prssanna committed:
fix: fix depends on ui test

Thunderbottom committed:
fix(requirements): remove psycopg2 from requirements

apparently, since version 2.8, psycopg2 does not install the binary
version by default (read source), and hence fails on setup with the
error:

Error: pg_config executable not found

since nobody can really be arsed to compile this binary on their own,
we'll stick to using psycopg2-binary instead.

source: https://www.postgresql.org/message-id/CA%2Bmi_8bd6kJHLTGkuyHSnqcgDrJ1uHgQWvXCKQFD3tPQBUa2Bw%40mail.gmail.com

Signed-off-by: Chinmay D. Pai <chinmaydpai@gmail.com>

ruchamahabal committed:
fix(Auto Repeat): derive next date from start date and offset (#9177)

surajshetty3416 committed:
Merge branch 'develop' into psycopg2-fix

surajshetty3416 committed:
Merge branch 'develop' into depends-on-test

mergify[bot] committed:
Merge pull request #9173 from prssanna/depends-on-test

test: fix depends on ui test

mergify[bot] committed:
Merge pull request #9175 from Thunderbottom/psycopg2-fix

fix(requirements): remove psycopg2 from requirements

surajshetty3416 committed:
Merge branch 'develop' into dev-webhook-tables

surajshetty3416 committed:
Merge branch 'develop' into bytes-fix

surajshetty3416 committed:
Merge branch 'develop' into awesome-bar

mergify[bot] committed:
Merge pull request #9154 from Thunderbottom/bytes-fix

fix(email): handle case where cstr returns text_type of str

surajshetty3416 committed:
Merge branch 'develop' into dev-webhook-tables

surajshetty3416 committed:
Merge branch 'develop' into qe_cf

mergify[bot] committed:
Merge pull request #9144 from scmmishra/awesome-bar

feat: wider awesomebar

surajshetty3416 committed:
Merge branch 'develop' into rename-doc-merge-alert

surajshetty3416 committed:
Merge branch 'develop' into email_ssl

mergify[bot] committed:
Merge pull request #9145 from hrwX/qe_cf

feat(Customize Form): Ability to add fields to quick entry

surajshetty3416 committed:
refactor: Commonify sent mail checks

-and fix formatting

mergify[bot] committed:
Merge pull request #9157 from sagarvora/fix-jinja-dev

fix: (re)allow custom filters/methods to be used with jinja

netchampfaris committed:
fix: Bug in show/hide global cards (#9163)



Co-authored-by: Suraj Shetty <13928957+surajshetty3416@users.noreply.github.com>

surajshetty3416 committed:
fix: Use _ for translation since frappe._ is not supported

surajshetty3416 committed:
fix:  Remove _ & _dict from frappe because add_module_properties ignores it

mergify[bot] committed:
Merge pull request #9180 from surajshetty3416/fix-energy-point-summary

fix: Use _ for translation since frappe._ is not supported

mergify[bot] committed:
Merge pull request #9164 from gavindsouza/rename-doc-merge-alert

feat: show_alert flag in rename_doc API

Thunderbottom committed:
fix: move email status check to fix travis

mail status wasn't being set when the check was run, so the output for
any() would always be false, and none of the mails would have status set
to sent in the backend. moving the check to a point after setting status
for each email should fix this issue

Signed-off-by: Chinmay D. Pai <chinmaydpai@gmail.com>

surajshetty3416 committed:
Merge branch 'develop' into dev-webhook-tables

mergify[bot] committed:
Merge pull request #9172 from Alchez/dev-webhook-tables

fix: allow tables to be sent with dates in webhook (develop)

mergify[bot] committed:
Merge pull request #9039 from Thunderbottom/email_ssl

feat(email): send over ssl and append outgoing mails to sent folder

mergify[bot] committed:
Merge pull request #9022 from Thunderbottom/chat-fix

fix(chat): refactor code to make chat work

mergify[bot] committed:
Merge pull request #9184 from gavindsouza/build-symlinks-fix

fix: bench build broken assets symlinks

msabukwaik committed:
fix: allow setting custom database schema in site_config (#9182)

* Update database.py

* Update frappe/database/postgres/database.py

Co-Authored-By: Chinmay Pai <chinmaydpai@gmail.com>

Co-authored-by: Chinmay Pai <chinmaydpai@gmail.com>

Thunderbottom commented:
@tukutela any updates on the tests?

tukutela commented:
Hey Chimnay,

Yes, I was able to get it passing the tests a while back.

On Fri, 3 Jan 2020 at 03:51, Chinmay Pai <notifications@github.com> wrote:

> @tukutela <https://github.com/tukutela> any updates on the tests?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/frappe/frappe/pull/8746?email_source=notifications&email_token=AACTGIEHFMEJU3MHOD4BUU3Q3YLPLA5CNFSM4JIJXMZKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEH6ZMLY#issuecomment-570267183>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AACTGIEGOOO35IUIOWVFTJDQ3YLPLANCNFSM4JIJXMZA>
> .
>


-- 
------------
Mark Walker
Maestro Results

Thunderbottom commented:
@tukutela we need a test to check what account it fetches as default, do you mind adding that?

also, you will have to rebase the branch since there is a conflict. let me know if you need any help.

Thunderbottom commented:
also, actually, we have fixed the issue with fetching the default email account in https://github.com/frappe/frappe/commit/6357a4a9a5aece037c618ec0b095cf90b15e6d45.

I suppose we could use that as a base and make some improvements to the logic, or close this pull request since the issue should be handled well by the system now.

tukutela commented:
ok, give me a few days to check it out, might help.

On Fri, 3 Jan 2020 at 15:30, Chinmay Pai <notifications@github.com> wrote:

> also, actually, we have fixed the issue with fetching the default email
> account in 6357a4a
> <https://github.com/frappe/frappe/commit/6357a4a9a5aece037c618ec0b095cf90b15e6d45>
> .
>
> I suppose we could use that as a base and make some improvements to the
> logic, or close this pull request since the issue should be handled well by
> the system now.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/frappe/frappe/pull/8746?email_source=notifications&email_token=AACTGIF6V4DZ55CCMX5V24LQ325ORA5CNFSM4JIJXMZKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEIAIEEI#issuecomment-570458641>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AACTGIEPO3JUESVTTJ2L4MLQ325ORANCNFSM4JIJXMZA>
> .
>


-- 
------------
Mark Walker
Maestro Results

Thunderbottom committed:
fix(email): safe_encode to avoid smtp ascii encoding issue

fixes issue where smtplib fails to encode the mail as ascii

Traceback (most recent call last):
  File "/home/frappe/frappe-bench/apps/frappe/frappe/app.py", line 60, in application
    response = frappe.api.handle()
  File "/home/frappe/frappe-bench/apps/frappe/frappe/api.py", line 55, in handle
    return frappe.handler.handle()
  File "/home/frappe/frappe-bench/apps/frappe/frappe/handler.py", line 22, in handle
    data = execute_cmd(cmd)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/handler.py", line 61, in execute_cmd
    return frappe.call(method, **frappe.form_dict)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/__init__.py", line 1042, in call
    return fn(*args, **newargs)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/core/doctype/user/user.py", line 801, in reset_password
    user.reset_password(send_email=True)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/core/doctype/user/user.py", line 234, in reset_password
    self.password_reset_mail(link)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/core/doctype/user/user.py", line 253, in password_reset_mail
    "password_reset", {"link": link}, now=True)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/core/doctype/user/user.py", line 298, in send_login_mail
    delayed=(not now) if now!=None else self.flags.delay_emails, retry=3)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/__init__.py", line 476, in sendmail
    inline_images=inline_images, header=header, print_letterhead=print_letterhead)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/email/queue.py", line 162, in send
    print_letterhead=print_letterhead)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/email/queue.py", line 185, in add
    send_one(email_queue.name, now=True)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/email/queue.py", line 475, in send_one
    raise e
  File "/home/frappe/frappe-bench/apps/frappe/frappe/email/queue.py", line 415, in send_one
    smtpserver.sess.sendmail(email.sender, recipient.recipient, message)
  File "/usr/lib64/python3.6/smtplib.py", line 855, in sendmail
    msg = _fix_eols(msg).encode('ascii')
UnicodeEncodeError: 'ascii' codec can't encode characters in position 335-339: ordinal not in range(128)

Signed-off-by: Chinmay D. Pai <chinmaydpai@gmail.com>

surajshetty3416 committed:
feat: Disable testing for all sites by default

To avoid accidental test execution on production sites

Thunderbottom committed:
chore: pass email_body test

Signed-off-by: Chinmay D. Pai <chinmaydpai@gmail.com>

mergify[bot] committed:
Merge pull request #9193 from surajshetty3416/disable-testing-by-default

chore: Disable testing for all sites by default

hrwX committed:
fix: unreferenced variable

mergify[bot] committed:
Merge pull request #9197 from hrwX/smtp-fx

fix(email): define _email_account before assignment

mergify[bot] committed:
Merge pull request #9189 from Thunderbottom/email-encode-fix

fix(email): safe_encode to avoid smtp ascii encoding issue

surajshetty3416 committed:
fix: Make background jobs logging quieter if quite flag is pass… (#9200)

surajshetty3416 committed:
fix: as_table msgprint (#9174)

* chore: improve readability

* fix: as_table msgprint

- fix as_table msgprint style and functionality
- Make table creation code more readable
- Make a text translatable

* style: Fix few deepsource issues

* style: Fix few deepsource issues

fix indentation

* fix: Incorrect assignment

* fix: Scrub URLs only for html format

* fix(typo): msg -> message

Co-authored-by: Raffael Meyer <meyer.raffael@gmail.com>

Commit:
attempting to resolve conflict

Commit:
added test feature to validate default is returned

Commit:
removing content

Thunderbottom commented:
did you rebase and push to this branch?

tukutela commented:
I had a look today, I think it makes sense to cancel this pull request,
there's a lot of updates to both smtp.py and queue.py (and the associated
doctypes) that aren't included within my original. I'll test again with the
new version in "develop" and start again from there.

On Sun, 5 Jan 2020 at 14:47, Chinmay Pai <notifications@github.com> wrote:

> did you rebase and push to this branch?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/frappe/frappe/pull/8746?email_source=notifications&email_token=AACTGIBJ6X3X63CLJAFHACDQ4FJ4LA5CNFSM4JIJXMZKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEIDHWWI#issuecomment-570850137>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AACTGIAUFJLJFAKSXRQBXI3Q4FJ4LANCNFSM4JIJXMZA>
> .
>


-- 
------------
Mark Walker
Maestro Results
